internal/cmd/manager/instance/upgrade/execute/cmd.go

```diff
@@ -277,6 +299,31 @@ func runInitDB(destDir string, walDir *string) error {
 		options = append(options, "--waldir", *walDir)
 	}
 
+	// Extract the WAL segment size from the pg_controldata output
+	walSegmentSizeString, ok := pgControlData[utils.PgControlDataBytesPerWALSegment]
+	if !ok {
+		return fmt.Errorf("no '%s' section into pg_controldata output", utils.PgControlDataBytesPerWALSegment)
+	}
+
+	walSegmentSize, err := strconv.Atoi(walSegmentSizeString)
+	if err != nil {
+		return fmt.Errorf(
+			"wrong '%s' pg_controldata value (not an integer): '%s' %w",
+			utils.PgControlDataBytesPerWALSegment, walSegmentSizeString, err)
+	}
+
+	options = append(options, "--wal-segsize="+strconv.Itoa(walSegmentSize/(1024*1024)))
+
+	// Extract the dat checksum version from the pg_controldata output
+	dataPageChecksumVersion, ok := pgControlData[utils.PgControlDataDataPageChecksumVersion]
+	if !ok {
+		return fmt.Errorf("no '%s' section into pg_controldata output", utils.PgControlDataDataPageChecksumVersion)
+	}
+
+	if dataPageChecksumVersion == "1" {
+		options = append(options, "--data-checksums")
+	}
```

Comment:
In postgres 18 we will have to set "--no-data-checksums" if checksums are disabled (they are enabled by default)